interactive_engine/executor/ir/graph_proxy/src/apis/graph/element/property.rs

```diff
@@ -44,40 +44,44 @@ impl PropKey {
         let prop_obj = if let PropKey::Len = self {
             element.len().into()
         } else {
-            let graph_element = element
-                .as_graph_element()
-                .ok_or_else(|| ExprEvalError::UnexpectedDataType(self.into()))?;
-            match self {
-                PropKey::Id => graph_element.id().into(),
-                PropKey::Label => graph_element
-                    .label()
-                    .map(|label| label.into())
-                    .unwrap_or(Object::None),
-                PropKey::Len => unreachable!(),
-                PropKey::All => graph_element
-                    .get_all_properties()
-                    .map(|obj| {
-                        obj.into_iter()
-                            .map(|(key, value)| {
-                                let obj_key: Object = match key {
-                                    NameOrId::Str(str) => str.into(),
-                                    NameOrId::Id(id) => id.into(),
-                                };
-                                (obj_key, value)
-                            })
-                            .collect::<Vec<(Object, Object)>>()
-                            .into()
-                    })
-                    .unwrap_or(Object::None),
-                PropKey::Key(key) => {
-                    if let Some(prop_val) = graph_element.get_property(key) {
-                        prop_val.try_to_owned().ok_or_else(|| {
-                            ExprEvalError::OtherErr("cannot get `Object` from `BorrowObject`".to_string())
-                        })?
-                    } else {
-                        Object::None
+            if let Some(graph_element) = element.as_graph_element() {
```

Comment:
Fix the implementation such that calling as_graph_element() from an None object can return a special object that have implementation of getting None value from any properties?

Reply:
Done. Introduced a `NullGraphEntry' for a null graph element.